src/snowflake/cli/_plugins/snowpark/snowpark_entity_model.py

```diff
@@ -59,17 +47,24 @@ class SnowparkEntityModel(EntityModelBase, ExternalAccessBaseModel, ImportsBaseM
         title="Python version to use when executing ", default=None
     )
     stage: str = Field(title="Stage in which artifacts will be stored")
-    artifacts: List[Union[PathMapping, str]] = Field(title="List of required sources")
+    artifacts: Artifacts = Field(title="List of required sources")
 
     @field_validator("artifacts")
     @classmethod
     def _convert_artifacts(cls, artifacts: Union[dict, str]):
         _artifacts = []
-        for artefact in artifacts:
-            if isinstance(artefact, PathMapping):
-                _artifacts.append(artefact)
+        for artifact in artifacts:
+            if (
+                "*" in artifact
```

Comment:
`*` isn't the only glob pattern, so that's a really weak check. What about something like `file[1-5].py` ?

Reply:
I wanted to to start with basic, I will try with glob.has_magic.

Reply:
I tried and it works after small changes!